src/core/utils/platform.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { getPlatform } from "./platform";
+import os from "os";
+
+describe("getPlatform", () => {
+  it("should return 'win-x64' when running on Windows", () => {
+    // Mock the os.platform function to return "win32"
+    jest.spyOn(os, "platform").mockReturnValue("win32");
+
+    // Call the function and expect it to return "win-x64"
+    expect(getPlatform()).toEqual("win-x64");
+  });
+
+  it("should return 'osx-x64' when running on macOS", () => {
+    // Mock the os.platform function to return "darwin"
+    jest.spyOn(os, "platform").mockReturnValue("darwin");
+
+    // Call the function and expect it to return "osx-x64"
+    expect(getPlatform()).toEqual("osx-x64");
+  });
+
+  it("should return 'linux-x64' when running on Linux", () => {
+    // Mock the os.platform function to return "linux"
+    jest.spyOn(os, "platform").mockReturnValue("linux");
+
+    // Call the function and expect it to return "linux-x64"
+    expect(getPlatform()).toEqual("linux-x64");
+  });
+
+  it("should return 'linux-x64' when running on Aix", () => {
+    // Mock the os.platform function to return "linux"
+    jest.spyOn(os, "platform").mockReturnValue("aix");
+
+    // Call the function and expect it to return "linux-x64"
+    expect(getPlatform()).toEqual("linux-x64");
+  });
+
+  it("should return 'linux-x64' when running on FreeBSD", () => {
+    // Mock the os.platform function to return "linux"
+    jest.spyOn(os, "platform").mockReturnValue("freebsd");
+
+    // Call the function and expect it to return "linux-x64"
+    expect(getPlatform()).toEqual("linux-x64");
+  });
+
+  it("should return 'linux-x64' when running on Sunos", () => {
+    // Mock the os.platform function to return "linux"
+    jest.spyOn(os, "platform").mockReturnValue("sunos");
```

Comment:
shall we also add one validation that this method was called with platform value ?

Reply:
added above